src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/ProjectDependencyHelper.cs

```diff
@@ -0,0 +1,109 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Runtime.Serialization;
+using Microsoft.CodeAnalysis.LanguageServer.LanguageServer;
+using Microsoft.CodeAnalysis.MSBuild;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.Extensions.Logging;
+using NuGet.ProjectModel;
+using NuGet.Versioning;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.HostWorkspace;
+internal static class ProjectDependencyHelper
+{
+    private const string UnresolvedProjectDependenciesName = "workspace/_roslyn_unresolvedProjectDependencies";
+
+    internal static bool HasUnresolvedDependencies(ProjectFileInfo projectFileInfo, ILogger logger)
+    {
+        var projectAssetsPath = projectFileInfo.ProjectAssetsFilePath;
+        if (!File.Exists(projectAssetsPath))
+        {
+            // If the file doesn't exist then all package references are unresolved.
+            logger.LogWarning(string.Format(LanguageServerResources.Project_0_has_unresolved_dependencies, projectFileInfo.FilePath));
```

Comment:
What about legacy projects? I don't think we'll have one at all in that case? Or projects still using packages.config?

Reply:
good point, I need to run this only for .net core projects.

For the latter case, I'm OK if we don't necessarily support that case (its not required for parity).